Coolguyzone/feat/monitors guide (#17433)
getsentry/sentry-docs: docs/guides/monitors.mdx

---
title: Setting Up Monitors
sidebar_order: 26

> description: "Practical guidance on choosing error, metric, cron, uptime, and mobile build monitors and avoiding false positives."

Reviewer: ```suggestion
description: "Practical guidance on when to use metric, cron, uptime, and mobile build monitors to customize issue creation thresholds."
```

early_access: true
---

> 

Reviewer: ```suggestion
<Include name="feature-available-for-user-group-early-adopter" />
```

^^ The EA alert

Monitors help you catch reliability problems early—anything from recurring errors crossing a threshold, to a drifting metric, a scheduled job that never ran, an HTTP probe that failed, or a mobile build regression.

> On the **[Monitors](https://sentry.io/monitors/)** page, the sidebar lists **All Monitors**, **My Monitors**, and filters for **Errors**, **Metrics**, **Crons**, **Uptime**, and **Mobile Builds**. **Alerts** is separate: monitors define conditions that produce [issues](/product/issues/); [alert rules](/guides/alerts/) decide how your team gets notified ([Custom Alerts](/product/new-monitors-and-alerts/alerts/) pair with monitors for routing and ticketing). See the [Monitors](/product/new-monitors-and-alerts/monitors/) product docs for Early Adopter access, creation flows, and settings.

Reviewer: ```suggestion
On the **[Monitors](https://sentry.io/monitors/)** page, you can filter by **All Monitors**, **My Monitors**, **Errors**, **Metrics**, **Crons**, **Uptime**, and **Mobile Builds**. While Monitors define conditions that create [issues](/product/issues/); [Alerts](/guides/alerts/) decide how your team gets notified. Pair ([Custom Alerts](/product/new-monitors-and-alerts/alerts/) with Monitors for notifying, ticketing, and more integration actions. See the [Monitors](/product/new-monitors-and-alerts/monitors/) product docs for creation flows and settings.
```

> Below is how those monitor categories map into Sentry. The deeper sections focus on [**cron**](#configuring-cron-monitors-well) and [**uptime**](#configuring-uptime-monitors-well) tuning—where small missteps most often drive false positives.

Reviewer: ```suggestion
Let's do a quick review of what each type of Monitor does before getting to some best practices.
```

## Monitor Types

Browse or filter monitors by category to match how they appear in the product:

> 

Reviewer: I suggest truncating all of this into a table that has the name of the monitor, a very brief description of what it does, and a suggestion on when to use it.

Author: Good suggestion! Incorporated.

### Errors

> [Error-focused monitors](/product/new-monitors-and-alerts/monitors/) customize when errors and related signals become issues—for example via Sentry’s default [**Error Monitor**](/product/new-monitors-and-alerts/monitors/#default-monitors) (rooted in [issue grouping](/concepts/data-management/event-grouping/)), or an **Issue Stream Monitor** that follows new issues across types.

Reviewer: ```suggestion
[Error Monitors](/product/monitors-and-alerts/monitors/#default-monitors) are the default monitors that come with setting up the [Sentry SDK](/platforms/). These are run by [issue grouping](/concepts/data-management/event-grouping/)).
```

Reviewer: Used as default Monitor.

### Metrics

> [**Metric monitors**](/product/new-monitors-and-alerts/monitors/#metric-monitor-settings) evaluate thresholds on data from errors, spans, logs, releases, and [application metrics](/product/explore/metrics/), using absolute thresholds, change detection, or dynamic anomaly detection. You can also open **Save As → Monitor for** from Application Metrics or **Create Monitor** from [Discover](/product/explore/discover-queries/) to start from an existing query.

Reviewer: ```suggestion
[**Metric Monitors**](/product/new-monitors-and-alerts/monitors/#metric-monitor-settings) evaluate thresholds on data from errors, spans, logs, releases, and application metrics, using absolute thresholds, change detection, or dynamic anomaly detection. You can also save Application Metrics using **Save As → Monitor for** or **Create a Monitor** from [Discover](/product/explore/discover-queries/) to start from an existing query.
```

Reviewer: Used as a way to monitor parts of your application that won't become issues through errors or performance.

### Crons

[**Cron monitors**](/product/new-monitors-and-alerts/monitors/crons/) track scheduled jobs: missed check-ins, late runs, and timeouts. Start from [cron monitor setup](/product/new-monitors-and-alerts/monitors/crons/getting-started/) in the Monitors workflow, or the dedicated [cron monitoring](/product/crons/) guides when instrumenting jobs with the SDK, CLI, or plain HTTP check-ins.

### Uptime

[**Uptime monitors**](/product/new-monitors-and-alerts/monitors/uptime-monitoring/) periodically probe HTTP endpoints for availability and failed responses—including [automatic detection](/product/new-monitors-and-alerts/monitors/uptime-monitoring/automatic-detection/) of common hostnames surfaced from your error traffic.

### Mobile Builds

**Mobile Builds** monitors track app size across iOS or Android builds and enforce thresholds (absolute size, diff, or relative diff)—for example regressions surfaced through [size analysis](/product/size-analysis/). See [Mobile Builds monitor settings](/product/new-monitors-and-alerts/monitors/#mobile-builds-monitor-settings) and [build distribution](/product/build-distribution/) for uploads and rollout context.

### Alerts

> **Alerts** in the sidebar is not another monitor flavor: it opens [alert rules and notifications](/product/new-monitors-and-alerts/alerts/) that react when monitors or existing issues satisfy your criteria. Start with [Setting Up Alerts](/guides/alerts/) for strategy and noise control.

Reviewer: ```suggestion
```
This doesn't seem at all necessary. We reference Alerts and pairing them with Monitors at the top

## What to Monitor

> You may be tempted to add monitors for everything. The better approach is to start with jobs and endpoints where a silent failure has a direct, delayed impact, allowing you to catch problems before they impact users.

Reviewer: I think we can have more about metric monitors here too. Some best practices about it, and a couple of quick examples could be really useful.

## Configuring Cron Monitors Well

A poorly configured cron monitor generates false positives. The key settings to get right:

### Schedule and Checkin Margin

The **schedule** tells Sentry when to expect a check-in. The **checkin margin** (grace period) is how long Sentry waits after a scheduled run before marking it missed.

Set the margin to account for:

- Scheduler startup time and job queue delay
- Variable runtime at the start (many jobs run quickly but occasionally hit slow paths)
- Timing mismatches between your job scheduler and Sentry's system

A margin of 5–10% of your job's expected interval is a reasonable starting point. For a job that runs every hour, a 5-minute margin is appropriate.

### Max Runtime (Timeout Detection)

Set **max runtime** to the longest your job should reasonably take. If the job sends an initial check-in but never sends a final one within this window, Sentry marks it as timed out.

Set this to your P99 runtime plus a buffer — not the average. A job that normally takes 30 seconds might occasionally take 3 minutes during high load. Setting max runtime to 5 minutes catches genuine hangs without false-firing on slow runs.

### Assign an Owner

> Assign an [owner](/product/crons/job-monitoring/#ownership) to each monitor — either a team or a specific person. Unowned monitors notify all project members, which is the fastest path to alert fatigue. The owner also receives escalation notifications when a monitor becomes broken.

Reviewer: This seems more generally useful! Let's not bury it in Crons.

Author: Good call, I moved it way up to the opening section.

### Monitor What Matters, Not Just `/health`

A `/health` endpoint that returns 200 OK but depends on no real application code can give you a false sense of security. Where possible, monitor an endpoint that exercises a real code path, even a lightweight one, like fetching a single record or verifying auth.

## Handling Noise

### Use Environments

Set alert rules for cron monitors to only fire for specific environments. A missed job in staging isn't worth a page — the same job missed in production is. [Configure environment filters](/product/crons/job-monitoring/#alerting-on-specific-environments) on your monitor alerts.

### Pause vs. Mute

**Pause** a monitor when you know the job won't run — during planned maintenance, deployments that disable a feature, or scheduled downtime. Pausing stops recording check-ins and stops notifications.

**Mute** a monitor when it's broken and you're not ready to fix it yet. Sentry will also automatically mute a monitor that has been consistently broken for 28 days to reduce notification noise.

Note: muting does not stop billing. To stop billing, deactivate or delete the monitor.

> ## Use Case Walkthroughs

Reviewer: These are all cron and uptime monitors, even in the mobile example, and we just introduce mobile build monitors. Let's make sure we have examples of at least one metric monitor in each of these use cases, and add a mobile build monitor to the mobile use case.

Reviewer: You'll also find that we'll want to reference much more on that table in terms of conditions, thresholds, etc for metric monitors. These are really powerful and absolutely should be highlighted.

---
title: Setting Up Monitors
sidebar_order: 26
description: "Practical guidance on when to use metric, cron, uptime, and mobile build monitors to customize issue creation thresholds."
early_access: true
---

Monitors help you catch reliability problems early—anything from recurring errors crossing a threshold, to a drifting metric, a scheduled job that never ran, an HTTP probe that failed, or a mobile build regression.

> On the **[Monitors](https://sentry.io/monitors/)** page, the sidebar lists **All Monitors**, **My Monitors**, and filters for **Errors**, **Metrics**, **Crons**, **Uptime**, and **Mobile Builds**. **Alerts** is separate: monitors define conditions that produce [issues](/product/issues/); [alert rules](/guides/alerts/) decide how your team gets notified ([Custom Alerts](/product/monitors-and-alerts/alerts/) pair with monitors for routing and ticketing). See the [Monitors](/product/monitors-and-alerts/monitors/) product docs for Early Adopter access, creation flows, and settings.

Reviewer: ```suggestion
On the **[Monitors](https://sentry.io/monitors/)** page, you can filter by **All Monitors**, **My Monitors**, **Errors**, **Metrics**, **Crons**, **Uptime**, and **Mobile Builds**. While Monitors define conditions that create [issues](/product/issues/); [Alerts](/guides/alerts/) decide how your team gets notified. Pair ([Custom Alerts](/product/new-monitors-and-alerts/alerts/) with Monitors for notifying, ticketing, and more integration actions. See the [Monitors](/product/new-monitors-and-alerts/monitors/) product docs for creation flows and settings.
```

Assign an [owner](/product/monitors-and-alerts/monitors/crons/job-monitoring/#ownership) to every monitor — either a team or a specific person. Unowned monitors notify all project members, which is the fastest path to alert fatigue. The owner also receives escalation notifications when a monitor becomes broken.

Let's do a quick review of what each type of Monitor does before getting to some best practices.

## Monitor Types

> | Type | What it does | When to use |

Reviewer: Love this format! So much easier to read.

| --- | --- | --- |

> | [Errors](/product/monitors-and-alerts/monitors/) | Controls when error events become issues | You want to customize how errors are grouped and surfaced |

Reviewer: ```suggestion
| [Errors](/product/monitors-and-alerts/monitors/) | Controls when error events become issues | Comes default with Sentry. Can be customized through Issue Grouping |
```

> | [Metrics](/product/monitors-and-alerts/monitors/#metric-monitor-settings) | Evaluates thresholds on errors, spans, logs, or [metrics](/product/explore/metrics/) using absolute, change, or anomaly detection | You need to alert on a quantitative signal over time |

Reviewer: ```suggestion
| [Metrics](/product/monitors-and-alerts/monitors/#metric-monitor-settings) | Evaluates thresholds on errors, spans, logs, or [application metrics](/product/explore/metrics/) using absolute, change, or anomaly detection | You need to alert on a quantitative signal over time |
```

| [Crons](/product/monitors-and-alerts/monitors/crons/) | Tracks scheduled jobs for missed check-ins, late runs, and timeouts | You have background jobs whose failures won’t surface as errors |
| [Uptime](/product/monitors-and-alerts/monitors/uptime-monitoring/) | Probes HTTP endpoints for availability on a set interval | You want to detect downtime on customer-facing or critical URLs |
| [Mobile Builds](/product/monitors-and-alerts/monitors/#mobile-builds-monitor-settings) | Tracks iOS and Android app size and enforces size thresholds | You want to catch binary size regressions before release |

> **Alerts** in the sidebar is not a monitor type: it opens [alert rules](/product/monitors-and-alerts/alerts/) that notify your team when monitors or issues satisfy your criteria. See [Setting Up Alerts](/guides/alerts/) for strategy and configuration.

Reviewer: ```suggestion
```
I just don't see the value in calling out Alerts right here. Let's find a better place to talk about using Alerts with Monitors if that feels important in this guide.

### Cron Jobs Worth Monitoring

Not every background job needs a monitor. Focus on jobs where a missed or failed run causes visible impact:

- **Revenue-critical**: Invoice generation, payment processing, subscription renewals
- **Data-critical**: Database backups, data exports, sync jobs with external systems
- **User-facing delivery**: Emails, push notification delivery, digest generation
- **Compliance or audit**: Log archival, GDPR deletion jobs, report generation

Lower priority: maintenance cleanup jobs, cache warming, analytics aggregation etc.

### Endpoints Worth Uptime Monitoring

Monitor endpoints at the level your users experience, not just your internal health checks:

- **Customer-facing entry points**: Login, signup, checkout, the homepage
- **Mobile/API entry points**: The base API endpoint your apps hit on launch
- **Webhooks you receive**: Payment processor callbacks, third-party integrations you depend on
- **Internal health checks**: Only if a failure there means something users would feel

Don't monitor every endpoint — that generates noise and makes it harder to act on real downtime.

### Metrics Worth Monitoring

> 

Reviewer: 👏

- **Account for deploys.** A spike immediately following a deployment is usually expected. If you ship frequently, consider a change-detection window that spans multiple days so normal deploy variance is included in the baseline.

## Configuring Uptime Monitors Well

### Start with Automatic Detection

Sentry [automatically configures an uptime monitor](/product/monitors-and-alerts/monitors/uptime-monitoring/automatic-detection/) for the most frequently seen hostname in your error data. Check this first — you may already have uptime monitoring for your primary domain enabled.

### Failure Threshold

By default, Sentry requires **three consecutive failures** before creating an uptime issue. This catches false positives from transient network issues or one-off timeouts. You can adjust this threshold when configuring the monitor, but the default is appropriate for most teams.

For internal services with high uptime expectations, consider lowering the threshold. For third-party endpoints you monitor but don't control, the default or higher is better.

### Monitor What Matters, Not Just `/health`

> A `/health` endpoint that returns 200 OK but depends on no real application code can give you a false sense of security. Where possible, monitor an endpoint that exercises a real code path, even a lightweight one, like fetching a single record or verifying auth.

Reviewer: ```suggestion
## Configuration Tips

The product docs cover full setup for each monitor type: [Cron](/product/monitors-and-alerts/monitors/crons/getting-started/), [Metric](/product/monitors-and-alerts/monitors/#metric-monitor-settings), and [Uptime](/product/monitors-and-alerts/monitors/uptime-monitoring/). Below are the opinionated shortcuts that could save you from learning the hard way.

### Cron Monitors

- **Set checkin margin to 5–10% of your job's interval.** A job that runs every hour should have a ~5-minute margin. This absorbs scheduler jitter and queue delay without hiding real misses.
- **Set max runtime to your P99 + buffer, not the average.** A job that normally takes 30 seconds might take 3 minutes under load. A 5-minute max runtime catches genuine hangs without false-firing on slow runs.
- **Link SDK errors to the monitor.** If you use the SDK's cron integration (`@monitor` decorator, `withMonitor`, etc.), errors are automatically associated via the monitor slug. If you use plain HTTP check-ins alongside a separately initialized SDK, set the monitor slug in your SDK context so errors surface on the monitor's detail page. See [Job Monitoring](/product/monitors-and-alerts/monitors/crons/job-monitoring/) for more.

### Metric Monitors

- **Start with absolute or change detection for critical metrics.** Anomaly detection needs historical data to build a baseline and can be slow to fire on novel regressions. Use it where traffic patterns make fixed thresholds impractical (weekday/weekend variation, seasonal traffic), not as a default.
- **Don't guess at thresholds.** Open the metric in [Discover](/product/explore/discover-queries/) or [Application Metrics](/product/explore/metrics/), review the last 30 days, and set the threshold just outside your worst legitimate spike. Revisit after two weeks — zero fires means too loose, multiple false positives means too tight or wrong detection type.
- **Match the evaluation window to the signal's cadence.** Error rate on a high-traffic endpoint: 5–10 minutes. p95 latency: 15–30 minutes. Crash-free session rate: 1–4 hours. Low-volume business metrics: 1–24 hours. Start longer and tighten once you know your baseline noise level.
- **One metric, one monitor.** A combined "health score" monitor is hard to act on. Separate monitors per metric make it obvious what's wrong.
- **Scope to production from the start.** Metric monitors on staging or development almost always generate noise.

### Uptime Monitors

- **Check automatic detection first.** Sentry [automatically configures an uptime monitor](/product/monitors-and-alerts/monitors/uptime-monitoring/automatic-detection/) for the most frequently seen hostname in your error data. You may already have one.
- **Keep the default failure threshold (3 consecutive failures).** This catches transient network blips and one-off timeouts. Lower it only for internal services with strict uptime expectations.
- **Monitor a real code path, not just `/health`.** A health endpoint that returns 200 without touching application code gives false confidence. Where possible, probe an endpoint that exercises a lightweight real path — fetching a record, verifying auth, etc.
```
This section currently keeps me from getting to some of the real goods below. I tried shortening it all to give us back a faster read. @coolguyzone what do you think?

Author: Yeah, I like your edit better too!

| Leaderboard recalculation      | Cron   | Schedule: every 5 min; max runtime: 3 min                                            | `#game-backend`     |
| Season reward distribution     | Cron   | Schedule: weekly (end of season); max runtime: 30 min                                | `#game-ops`         |
| Matchmaking queue cleanup      | Cron   | Schedule: every 1 min; max runtime: 30 sec                                           | `#matchmaking-team` |
| Server asset sync              | Cron   | Schedule: nightly; max runtime: 20 min                                               | `#infra`            |
| Game API entry point           | Uptime | Probe: every 1 min; failure threshold: 3 consecutive                                 | `#game-oncall`      |
| Match results webhook receiver | Uptime | Probe: every 5 min; failure threshold: 3 consecutive                                 | `#game-backend`     |
| Crash-free session rate        | Metric | Signal: crash-free sessions; detection: change; threshold: drops > 5% in 1h window  | `#game-oncall`      |

> | Matchmaking error rate         | Metric | Signal: error rate on matchmaking spans; detection: absolute; threshold: > 2% in 5 min | `#matchmaking-team` |

Reviewer: ```suggestion
### Gaming

A multiplayer game backend runs scheduled jobs that affect player experience directly: leaderboard updates, season reward distribution, matchmaking queue cleanup. Silent failures show up as stale data or missing rewards, as support tickets, not error alerts. Server-side crashes and matchmaking degradation need metric-level detection.

**Cron Monitors:**

| Monitor                    | Schedule                                   | Grace Period | Max Runtime | Failure Tolerance | Owner               |
| -------------------------- | ------------------------------------------ | ------------ | ----------- | ----------------- | ------------------- |
| Leaderboard recalculation  | Interval: every 5 minutes                  | 1 min        | 3 min       | 1                 | `#game-backend`     |
| Season reward distribution | Crontab: `0 0 * * 0` (weekly)             | 5 min        | 30 min      | 1                 | `#game-ops`         |
| Matchmaking queue cleanup  | Interval: every 1 minute                   | 15 sec       | 30 sec      | 2                 | `#matchmaking-team` |
| Server asset sync          | Crontab: `0 3 * * *` (nightly at 3 AM)    | 5 min        | 20 min      | 1                 | `#infra`            |

**Metric Monitors:**

| Monitor                  | Dataset  | Aggregate                 | Detection | Threshold / Comparison                                                   | Interval | Environment  | Owner               |
| ------------------------ | -------- | ------------------------- | --------- | ------------------------------------------------------------------------ | -------- | ------------ | -------------------- |
| Crash-free session rate  | Releases | `crash_free_rate(session)` | Change   | Below, High: 5%, compared to previous 1 day                              | 1 hour   | `production` | `#game-oncall`       |
| Matchmaking error rate   | Errors   | `count()`                 | Threshold | Above, High: 50 errors; query: `transaction:/api/matchmaking/*`          | 5 min    | `production` | `#matchmaking-team`  |

**Uptime Monitors:**

| Monitor                        | URL                                        | Method | Interval   | Failure Threshold | Environment  | Owner           |
| ------------------------------ | ------------------------------------------ | ------ | ---------- | ----------------- | ------------ | --------------- |
| Game API entry point           | `https://api.game.example.com/v1/profile`   | GET    | Every 1 min | 3                | `production` | `#game-oncall`  |
| Match results webhook receiver | `https://api.game.example.com/webhooks/match` | POST | Every 5 min | 3                | `production` | `#game-backend` |
```

I tried a different format for this, and referenced the terms used in product so that it felt more 1:1 in terms of configuration if someone wanted to copy elements or duplicate. Would love your thoughts @coolguyzone

> **Noise strategy:** Tag each job with `game_region` and configure separate monitor environments per region (us-east, eu-west, ap-southeast). A failed leaderboard job in one region is a localized issue; failed in all three at once means something systemic. For the metric monitors, use a 1-hour window on crash-free rate to smooth out transient session drops — a 5-minute spike is noise, a sustained drop is a release regression.

Reviewer: ```suggestion
**Noise strategy:** Set the matchmaking queue cleanup to **Failure Tolerance: 2** — a single missed check-in on a 1-minute job is often scheduler jitter, not a real failure. For the crash-free rate metric monitor, use **Change** detection with a 1-hour interval to smooth out transient session drops; a 5-minute spike is noise, a sustained drop relative to the previous day is a release regression. Tag cron jobs with `game_region` and use separate monitor environments per region so a failed leaderboard job in one region doesn't mask a systemic outage across all three.
```
Just added a bit more flavor with the additional details

### SaaS

> SaaS products depend on a pipeline of billing, notification, and data jobs all running reliably. A missed billing job means invoices don't go out; a failed data export job means a customer's Monday workflow breaks. Monitors can be used to catch these issues right away.

Reviewer: ```suggestion
SaaS products depend on a pipeline of billing, notification, and data jobs all running reliably. A missed billing job means invoices don't go out; a failed data export job means a customer's Monday workflow breaks. Metric monitors on checkout error rates and API latency catch degradation before it becomes an outage.
```

| Invoice generation           | Cron   | Schedule: daily (end of billing period); max runtime: 15 min                                         | `#billing-eng`  |
| Payment retry processing     | Cron   | Schedule: every 6 hours; max runtime: 10 min                                                         | `#billing-eng`  |
| Trial expiration enforcement | Cron   | Schedule: daily; max runtime: 5 min                                                                  | `#growth-eng`   |
| Scheduled data exports       | Cron   | Schedule: varies per customer; max runtime: 30 min                                                   | `#data-eng`     |
| Notification delivery job    | Cron   | Schedule: every 2 min; max runtime: 90 sec                                                           | `#platform-eng` |
| API base endpoint            | Uptime | Probe: every 1 min; failure threshold: 3 consecutive                                                 | `#oncall`       |
| Payment error rate           | Metric | Signal: error rate on checkout/billing spans; detection: absolute; threshold: > 1% in 5 min window  | `#billing-eng`  |

> | API p95 latency              | Metric | Signal: p95 response time; detection: change; threshold: increases > 20% vs. previous 24h           | `#oncall`       |

Reviewer: ```suggestion
**Cron Monitors:**

| Monitor                      | Schedule                                    | Grace Period | Max Runtime | Failure Tolerance | Owner           |
| ---------------------------- | ------------------------------------------- | ------------ | ----------- | ----------------- | --------------- |
| Invoice generation           | Crontab: `0 23 * * *` (daily at 11 PM)     | 10 min       | 15 min      | 1                 | `#billing-eng`  |
| Payment retry processing     | Interval: every 6 hours                     | 5 min        | 10 min      | 1                 | `#billing-eng`  |
| Trial expiration enforcement | Crontab: `0 6 * * *` (daily at 6 AM)       | 5 min        | 5 min       | 1                 | `#growth-eng`   |
| Scheduled data exports       | Interval: varies per customer               | 10 min       | 30 min      | 1                 | `#data-eng`     |
| Notification delivery        | Interval: every 2 minutes                   | 30 sec       | 90 sec      | 2                 | `#platform-eng` |

**Metric Monitors:**

| Monitor              | Dataset | Aggregate                 | Detection | Threshold / Comparison                                                                 | Interval | Environment  | Owner          |
| -------------------- | ------- | ------------------------- | --------- | -------------------------------------------------------------------------------------- | -------- | ------------ | -------------- |
| Payment error rate   | Spans   | `failure_rate()`          | Threshold | Above, High: 1%; query: `span.op:http transaction:/api/checkout/*`                     | 5 min    | `production` | `#billing-eng` |
| API p95 latency      | Spans   | `p95(span.duration)`      | Change    | Above, High: 20%, compared to previous 1 day                                           | 15 min   | `production` | `#oncall`      |

**Uptime Monitors:**

| Monitor            | URL                                         | Method | Interval    | Failure Threshold | Environment  | Owner    |
| ------------------ | ------------------------------------------- | ------ | ----------- | ----------------- | ------------ | -------- |
| API base endpoint  | `https://api.example.com/v1/me`  | GET | Every 1 min | 3 | `production` | `#oncall` |
```
Same with this one - trying a different format.

> **Noise strategy:** Add a `checkin_margin` generous enough for the billing job to handle end-of-month volume spikes. Use a separate monitor environment for `production`, `staging`, and `eu` (if you run a separate EU instance) and only alert on `production`. For the payment error rate metric monitor, use an absolute threshold rather than change detection — even a small baseline error rate on checkout is unacceptable, so percentage-change detection would set the wrong expectation.

Reviewer: ```suggestion
**Noise strategy:** Set a generous **Grace Period** on the invoice generation job to handle end-of-month volume spikes — 10 minutes instead of the default 1. Use separate monitor environments for `production` and `eu` (if you run a separate EU instance) and only connect alerts to production monitors. For the payment error rate monitor, use **Threshold** (absolute) detection rather than **Change**. Even a small baseline error rate on checkout is unacceptable, so percentage-change detection would set the wrong expectation.
```

### Mobile

> Mobile apps run backend jobs that affect users indirectly: push notification delivery, app store review sync, analytics aggregation, and device registration cleanup. When these fail silently, users see stale content or stop receiving notifications. Build size regressions can also bloat download size and tank conversion at install time. Monitors can catch all of these before users are affected.

Reviewer: ```suggestion
Mobile apps run backend jobs that affect users indirectly: push notification delivery, app store review sync, analytics aggregation, and device registration cleanup. When these fail silently, users see stale content or stop receiving notifications. Metric monitors on crash-free rates per platform catch release regressions, and mobile build monitors catch binary size bloat before it reaches the app store, or impacts users.
```

| Analytics aggregation            | Cron          | Schedule: hourly; max runtime: 15 min                                                                      | `#data-eng`        |
| Mobile API base endpoint         | Uptime        | Probe: every 1 min; failure threshold: 3 consecutive                                                       | `#mobile-oncall`   |
| Push notification service health | Uptime        | Probe: every 5 min; failure threshold: 3 consecutive                                                       | `#mobile-backend`  |
| Crash-free session rate (iOS)    | Metric        | Signal: crash-free sessions, iOS; detection: change; threshold: drops > 3% vs. previous 24h               | `#mobile-oncall`   |
| Crash-free session rate (Android)| Metric        | Signal: crash-free sessions, Android; detection: change; threshold: drops > 3% vs. previous 24h           | `#mobile-oncall`   |
| App binary size (iOS)            | Mobile Build  | Platform: iOS; threshold: size diff > 3 MB per build                                                      | `#mobile-platform` |

> | App binary size (Android)        | Mobile Build  | Platform: Android; threshold: size diff > 2 MB per build                                                  | `#mobile-platform` |

Reviewer: ```suggestion
**Cron Monitors:**

| Monitor                    | Schedule                                  | Grace Period | Max Runtime | Failure Tolerance | Owner              |
| -------------------------- | ----------------------------------------- | ------------ | ----------- | ----------------- | ------------------ |
| Push notification delivery | Interval: every 1 minute                  | 15 sec       | 45 sec      | 2                 | `#mobile-backend`  |
| App store review sync      | Interval: every 4 hours                   | 5 min        | 10 min      | 1                 | `#mobile-platform` |
| Device token cleanup       | Crontab: `0 2 * * 0` (weekly at 2 AM)    | 5 min        | 20 min      | 1                 | `#mobile-backend`  |
| Analytics aggregation      | Interval: every 1 hour                    | 5 min        | 15 min      | 1                 | `#data-eng`        |

**Metric Monitors:**

| Monitor                       | Dataset  | Aggregate                  | Detection | Threshold / Comparison                                    | Interval | Environment  | Owner             |
| ----------------------------- | -------- | -------------------------- | --------- | --------------------------------------------------------- | -------- | ------------ | ----------------- |
| Crash-free sessions (iOS)     | Releases | `crash_free_rate(session)` | Change    | Below, High: 3%, compared to previous 1 day               | 1 hour   | `production` | `#mobile-oncall`  |
| Crash-free sessions (Android) | Releases | `crash_free_rate(session)` | Change    | Below, High: 3%, compared to previous 1 day               | 1 hour   | `production` | `#mobile-oncall`  |

**Uptime Monitors:**

| Monitor                          | URL                                              | Method | Interval    | Failure Threshold | Environment  | Owner             |
| -------------------------------- | ------------------------------------------------ | ------ | ----------- | ----------------- | ------------ | ----------------- |
| Mobile API base endpoint         | `https://mobile-api.example.com/v1/bootstrap`    | GET    | Every 1 min | 3                 | `production` | `#mobile-oncall`  |
| Push notification service health | `https://push.example.com/v1/me`                | GET    | Every 5 min | 3                 | `production` | `#mobile-backend` |

**Mobile Build Monitors:**

| Monitor                  | Measurement  | Threshold Type | High Priority | Low Priority | Owner              |
| ------------------------ | ------------ | -------------- | ------------- | ------------ | ------------------ |
| App binary size (iOS)    | Install Size | Absolute Diff  | 3 MB          | 1 MB         | `#mobile-platform` |
| App binary size (Android)| Install Size | Absolute Diff  | 2 MB          | 0.5 MB       | `#mobile-platform` |
```

Same idea, trying a different format

> **Noise strategy:** Push notification delivery runs frequently and is sensitive to queue depth — add a `checkin_margin` of 15 seconds to absorb burst delays. Separate the crash-free rate metric monitors by platform (iOS and Android) rather than combining them: a crash spike on one platform is almost always release- or device-specific, and a combined monitor would obscure which platform is affected. Set the mobile build size thresholds conservatively at first and tighten them after a few releases — you want to catch regressions, not block every PR that adds a dependency.

Reviewer: ```suggestion
**Noise strategy:** Push notification delivery runs frequently and is sensitive to queue depth — add a `checkin_margin` of 15 seconds to absorb burst delays. Separate the crash-free rate metric monitors by platform (iOS and Android) rather than combining them: a crash spike on one platform is almost always release- or device-specific, and a combined monitor would obscure which platform is affected. Set mobile build size thresholds conservatively at first using **Absolute Diff** (size change from previous build) rather than absolute values — you want to catch regressions, not block every PR that adds a dependency. Tighten after a few releases.
```
